tools/src/keygen/KeygenOptions.hs

```diff
@@ -29,15 +31,18 @@ data KeygenOptions = KeygenOptions
     , koConfigurationOptions :: ConfigurationOptions
     } deriving (Show)
 
+newtype DgdPath      = DgdPath FilePath deriving (Show)
+
+newtype DgdCanonical = DgdCanonical Bool deriving (Show)
+
 data KeygenCommand
     = RearrangeMask FilePath
     | GenerateKey FilePath
     | GenerateVss FilePath
     | ReadKey FilePath
     | DumpAvvmSeeds DumpAvvmSeedsOptions
     | GenerateKeysBySpec GenKeysOptions
-    | DumpGenesisData { dgdPath      :: !FilePath
-                      , dgdCanonical :: !Bool }
+    | DumpGenesisData DgdPath DgdCanonical
```

Comment:
I see that you have `newtype`ed these. In this case I actually would not bother. Newtypes are useful when you have two constructor fields of the same type. like:
```
    | Command Bool Bool!
```
which I would usually `newtype` as:
```
newtype OpenDoors = OpenDoors Bool
newtype LaunchMissles = LauchMissiles Bool

data X
     = Command OpenDoors LauchMissiles
```
This means you reduces the chances of getting the two `Bool` fields around the wrong way.

So in the example above, I would be perfectly happy with:
```
    | DumpGenesisData !FilePath !Bool
```
because the compiler will not *let* anyone get it wrong.

Reply:
Yes, that's true :) The primary advantage of a `newtype` here is that you can provide some additional meaning to the type, track provenance, not use it incorrectly, etc. For this, it is likely of small benefit.